openlibrary/plugins/worksearch/code.py

```diff
@@ -217,48 +216,43 @@ def process_individual_sort(sort):
     return ','.join(process_individual_sort(s.strip()) for s in raw_sort.split(','))
 
 
-def read_author_facet(af):
-    # example input: "OL26783A Leo Tolstoy"
-    return re_author_facet.match(af).groups()
+def read_author_facet(author_facet: str) -> tuple[str, str]:
+    key, name = author_facet.split(' ', 1)
```

Comment:
Optional: Remove leading and trailing whitespace...
```suggestion
    key, name = author_facet.strip().split(' ', 1)
```

Reply:
Add the doctest :+1: There shouldn't be a need to strip here; the value coming from solr should not have any leading whitespace :+1: